client/components/button/index.jsx

```diff
@@ -3,49 +3,43 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import assign from 'lodash/assign';
+import { PropTypes, createElement, PureComponent } from 'react';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
-
-export default React.createClass( {
-
-	displayName: 'Button',
-
-	propTypes: {
-		disabled: React.PropTypes.bool,
-		compact: React.PropTypes.bool,
-		primary: React.PropTypes.bool,
-		scary: React.PropTypes.bool,
-		type: React.PropTypes.string,
-		href: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		borderless: React.PropTypes.bool
-	},
-
-	getDefaultProps() {
-		return {
-			disabled: false,
-			type: 'button',
-			onClick: noop,
-			borderless: false
-		};
-	},
+import { omit } from 'lodash';
+
+export default class Button extends PureComponent {
+	static propTypes = {
+		compact: PropTypes.bool,
+		primary: PropTypes.bool,
+		scary: PropTypes.bool,
+		type: PropTypes.string,
+		href: PropTypes.string,
+		borderless: PropTypes.bool
+	};
+
+	static defaultProps = {
+		type: 'button'
+	};
 
 	render() {
-		const element = this.props.href ? 'a' : 'button';
-		const buttonClasses = classNames( {
-			button: true,
-			'is-compact': this.props.compact,
-			'is-primary': this.props.primary,
-			'is-scary': this.props.scary,
-			'is-borderless': this.props.borderless
+		const omitProps = [ 'compact', 'primary', 'scary', 'borderless' ];
+
+		let tag;
+		if ( this.props.href ) {
+			tag = 'a';
+			omitProps.push( 'type' );
```

Comment:
>Yeah, I used Babel to see its output and it uses `concat`.

Got me curious, so I checked; doesn't appear to affect the `push` in my testing, and I wouldn't really expect it to.

```js
(0, _createClass3.default)(Button, [{
	key: 'render',
	value: function render() {
		var omitProps = ['compact', 'primary', 'scary', 'borderless'];

		var tag = void 0;
		if (this.props.href) {
			tag = 'a';
			omitProps.push('type');
		} else {
			tag = 'button';
		}

		return (0, _react.createElement)(tag, (0, _extends3.default)({}, (0, _omit3.default)(this.props, omitProps), {
			className: (0, _classnames2.default)('button', this.props.className, {
				'is-compact': this.props.compact,
				'is-primary': this.props.primary,
				'is-scary': this.props.scary,
				'is-borderless': this.props.borderless
			})
		}));
	}
}]);
```